test/jdk/java/util/zip/ZipFile/TestZipFileEncodings.java

```diff
@@ -115,6 +124,54 @@ public void testUnicodeManyEntries(String charsetName) throws Throwable {
         test(70000, 10, true, Charset.forName(charsetName));
     }
 
+    @Test
+    public void latin1NotAscii() throws IOException {
+
+        Path zip = Path.of("latin1-not-ascii.zip");
+
+        // latin1, but not ASCII
+        String entryName = "smörgåsbord";
+
+        try (ZipOutputStream z = new ZipOutputStream(Files.newOutputStream(zip))) {
+            z.putNextEntry(new ZipEntry(entryName));
+        }
+
+        try (ZipFile z = new ZipFile(zip.toFile())) {
+            assertNotNull(z.getEntry(entryName));
+        }
+    }
+    @Test(dataProvider = "all-charsets")
```

Comment:
Please add a comment introducing the test

Reply:
I described the rationale of adding this test in a comment.